src/tasks/setApprovals.ts

```diff
@@ -0,0 +1,116 @@
+import { promises as fs } from "fs";
+
+import "@nomiclabs/hardhat-ethers";
+import { Contract } from "ethers";
+import { task, types } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { SettlementEncoder } from "../ts";
+
+import { createGasEstimator, gweiToWei } from "./ts/gas";
+import { prompt } from "./ts/tui";
+
+interface Approval {
+  spender: string;
+  token: string;
+}
+
+interface Args {
+  input: string;
+  dryRun: boolean;
+  gasInGwei: number;
+}
+
+async function setApprovals(
+  { input, dryRun, gasInGwei }: Args,
+  hre: HardhatRuntimeEnvironment,
+) {
+  // Instantiate Settlement Contract
+  const settlementDeployment = await hre.deployments.get("GPv2Settlement");
+  const settlement = new Contract(
+    settlementDeployment.address,
+    settlementDeployment.abi,
+  ).connect(hre.ethers.provider);
+  const [signer] = await hre.ethers.getSigners();
+
+  //Instantiate ERC20 ABI
+  const IERC20 = await hre.artifacts.readArtifact(
+    "src/contracts/interfaces/IERC20.sol:IERC20",
+  );
+  const token = new Contract(
+    hre.ethers.constants.AddressZero,
+    IERC20.abi,
+    hre.ethers.provider,
+  );
```

Comment:
ubernit:
```suggestion
  const token = new ethers.utils.Interaction(IERC20.abi);
```

Then you can call `encodeFunctionData` directly:
```ts
token.encodeFunctionData(...)
```